metadata_backend/api/operators/file.py

```diff
@@ -0,0 +1,259 @@
+                    "_id": 0,
+                    "name": 1,
+                    "path": 1,
+                    "project": 1,
+                    "bytes": "$versions.bytes",
+                    "encrypted_checksums": "$versions.encrypted_checksums",
+                    "unencrypted_checksums": "$versions.unencrypted_checksums",
+                }
+            },
+        ]
+        if version:
+            # get a specific version
+            aggregate_query.insert(2, {"$match": {"versions.version": version}})
+        else:
+            # sort to get the latest version
+            aggregate_query.insert(2, {"$sort": {"versions.version": -1}})
+            # get only the latest version
+            aggregate_query.insert(3, {"$limit": 1})
+        try:
+            file = await self.db_service.do_aggregate("file", aggregate_query)
+        except (ConnectionFailure, OperationFailure) as error:
+            reason = f"Error happened while getting file, err: {error}"
+            LOG.exception(reason)
+            raise web.HTTPInternalServerError(reason=reason) from error
+        if not file:
+            reason = f"File '{accession_id}' was not found."
+            LOG.error(reason)
+            raise web.HTTPNotFound(reason=reason)
+
+        return file[0]
+
+    async def read_project_files(self, project_id: str) -> List[Dict]:
+        """Read files from DB based on a specific filter type and corresponding identifier.
+
+        The files are read by the latest version, and filtered either by projectId.
+
+        :param project_id: Project ID to get files for
+        :returns: List of files
+        """
+        aggregate_query = [
+            {"$match": {"projectId": project_id}},
+            {"$sort": {"versions.version": -1}},
+            {"$unwind": "$versions"},
+            {
+                "$project": {
+                    "_id": 0,
+                    "name": 1,
+                    "path": 1,
+                    "project": 1,
+                    "bytes": "$version.bytes",
+                    "encrypted_checksums": "$version.encrypted_checksums",
+                    "unencrypted_checksums": "$version.unencrypted_checksums",
+                }
+            },
+        ]
+        try:
+            return await self.db_service.do_aggregate("file", aggregate_query)
+        except (ConnectionFailure, OperationFailure) as error:
+            reason = f"Error happened while getting files for project {project_id}, err: {error}"
+            LOG.exception(reason)
+            raise web.HTTPInternalServerError(reason=reason) from error
+
+    async def read_submission_files(self, submission_id: str) -> List[Dict]:
+        """Get files in a submission.
+
+        The files are identified in a submission by version.
+
+        :param submission_id: Submission ID to get files for
+        :returns: List of files specific to a submission
+        """
+        # TO_DO: add check for ready status
```

Comment:
Was this dealt with or does it need a further issue?

Reply:
part of #148